### ISSUE TITLE
`flk update` fails because of obsolete flag

### ISSUE THREAD

**Issue by lourkeur** [divnix/devos#159]
## Expected Behavior
`flk update` goes smoothly

## Current Behavior
```
[devshell]$ flk update
error: unrecognised flag '--recreate-lock-file'
Try 'nix --help' for more information.
```

## Possible Solution
remove all 2 occurrences of this CLI flag

## Steps to Reproduce
1. clone/get devos
2. Run `flk update` to get nix to the latest revision
3. Restart the devShell
4. Run `flk update`

## Context
Went to `flk update` this morning on https://github.com/lourkeur/config (based off 0.7) and it didn't work. This also affects trunk.

## Your Environment
NixOS 20.09 x86_64

**Comment by nrdxp** [comment]
This flag still exists in the nixFlakes from nixpkgs master, which is what users will use by default. The latest nix from nix master has changed the semantics of the `update` command to make the `--recreate-lock-file` obsolete. 

There have been quite a number of useful changes to Nix in the last few months, including improved documentation, so what I was planning on doing very soon, was to set the system nix from the `nix` flake instead of from `nixpkgs` so that users can enjoy all the latest changes. Once that's done, this will be resolved.

**Comment by Pacman99** [comment]
Also if you update your flake, `nix flake check` will be failing for another day or so, because of a bug with aarch64 nixUnstable that results in it depending libcpuid, which doesn't evaluate for aarch64. The fix has been [backported](https://github.com/NixOS/nixpkgs/pull/116083), but hydra hasn't gotten to it yet.

**Comment by nrdxp** [comment]
fixed by d51cd34fb7542bc6d4baf1e9393b7c1e90d5806b
